### ISSUE TITLE
Only under certain conditions can filter_count be returned.

### ISSUE THREAD

**Issue by cdwmhcc** [directus/api#1074]
# Bug Report
Sometimes you can return filter_count, sometimes you can't.

##### 1、
``` javascript
let searchkey = ""
let categories = ""

this.client.getItems("blog",{
    meta:"total_count,result_count,filter_count",
    sort: "-created_on",
    filter:{
        "category.id": {
            in:  category 
         },
         title:{
             logical: "or",
             contains: searchkey 
         },
         description:{
             logical: "or",
             contains: searchkey 
         },
     },
     limit: 10,
     fields: "*.*",
     status: "published"
})
```

At this point, because searchkey and categories are empty, no filter is passed, and filter_count cannot be returned.

![image](https://user-images.githubusercontent.com/8030134/60485550-f68b5300-9ccf-11e9-8bcd-7b4fc1ebc585.png)

![image](https://user-images.githubusercontent.com/8030134/60485177-9e078600-9cce-11e9-84b0-db32bb95d282.png)

#####  2、

``` javascript
let searchkey = ""
let categories = "7,13,15,8,14,32"

this.client.getItems("blog",{
    meta:"total_count,result_count,filter_count",
    sort: "-created_on",
    filter:{
        "category.id": {
            in:  category 
         },
         title:{
             logical: "or",
             contains: searchkey 
         },
         description:{
             logical: "or",
             contains: searchkey 
         },
     },
     limit: 10,
     fields: "*.*",
     status: "published"
})
```

At this point, because the search key is empty and the categories have a value, the filter [category. id] [in]: 7, 13, 15, 8, 14, 32 is passed, and the filter_count is returned correctly.

![image](https://user-images.githubusercontent.com/8030134/60485703-6e597d80-9cd0-11e9-9557-a6d9f670231d.png)

![image](https://user-images.githubusercontent.com/8030134/60487820-d9a44f00-9cd2-11e9-809a-35aa906a3f9d.png)

##### 3、

``` javascript
let searchkey = ""
let categories = "28,29,30"

this.client.getItems("blog",{
    meta:"total_count,result_count,filter_count",
    sort: "-created_on",
    filter:{
        "category.id": {
            in:  category 
         },
         title:{
             logical: "or",
             contains: searchkey 
         },
         description:{
             logical: "or",
             contains: searchkey 
         },
     },
     limit: 10,
     fields: "*.*",
     status: "published"
})
```
The filter [category. id] [in]: 7, 13, 15, 8, 14, 32 is passed, and no data is returned after filtering, so filter_count cannot be returned at this time.
![image](https://user-images.githubusercontent.com/8030134/60487582-84683d80-9cd2-11e9-9ecc-f9109246ecab.png)

![image](https://user-images.githubusercontent.com/8030134/60487779-be394400-9cd2-11e9-9e6f-039a12607b9f.png)

## Technical Details

- Device: [eg: Desktop]
- OS: [Windows 10.1809]
- Web Server: [Nginx:1.17.0]
- PHP Version: [7.3.6]
- Database: [5.7.26]
- API Version: [2.2.1]
- APP Version: [7.6.1]
- Install Method: [cloned `master` branch]

**Comment by cdwmhcc** [comment]
I'm doing this now, but it's not a good solution.

``` javascript
let searchkey = "", categories = "", pagesize = 10, sort = '-created_on'
this.client.getItems("blog",{
    meta:"total_count,result_count,filter_count",
    sort:  sort,
    filter:{
        "category.id": {
            in:  category 
         },
         title:{
             logical: "or",
             contains: searchkey 
         },
         description:{
             logical: "or",
             contains: searchkey 
         },
     },
     limit: pagesize,
     fields: "*.*",
     status: "published"
}).then((res) => {
    let filter_total = 0
    if(softlist.meta.result_count === 0){
        filter_total = 0
    }else if(softlist.meta.result_count > 0){
        if(softlist.meta.filter_count){
            filter_total = softlist.meta.filter_count
        }else {
            filter_total = softlist.meta.total_count
        }
    }
})
```

**Comment by bjgajjar** [comment]
Fixed in #1088 1088

**Comment by cdwmhcc** [comment]
> Fixed in #1088

Thank you very much :+1::+1::+1:
